src/main/java/org/intellij/sdk/codesync/clients/JSONResponse.java

```diff
@@ -78,7 +79,8 @@ public void raiseForStatus () throws StatusCodeError {
                     errorMessage = (String) errorObject.get("message");
                 }
                 if (errorObject != null && errorObject.containsKey("error_code")) {
-                    customErrorCode = ((Number) errorObject.get("error_code")).intValue();
+                    errorCode = ((Number) errorObject.get("error_code")).intValue();
```

Comment:
@saadberry I do not think this will fix this issue. You should add a null value check in here. Something like

```

Number errorCode = (Number) errorObject.get("error_code");

if (errorCode != null){
    customErrorCode = errorCode.intValue();
}
```

Reply:
But my other question would be, why do we want to convert it to integer? can't we directly use `Number` object?

Reply:
That's true, updated me code to check value before assining 



> @saadberry I do not think this will fix this issue. You should add a null value check in here. Something like
> 
> ```
> 
> Number errorCode = (Number) errorObject.get("error_code");
> 
> if (errorCode != null){
>     customErrorCode = errorCode.intValue();
> }
> ```

Reply:
> But my other question would be, why do we want to convert it to integer? can't we directly use `Number` object?

![image](https://github.com/codesyncapp/codesync-intellij/assets/15974895/475eb7c7-7a27-4ceb-a7c1-d3bf00d812e6)

I believe we need to pass int param as that's what our method is expecting